src/Task.h

```diff
@@ -617,6 +617,26 @@ class Task {
    */
   std::string syscall_name(int syscallno) const;
 
+  /**
+   * Look up a TLS address for this thread.  |offset| and
+   * |load_module| are as specified in the qGetTLSAddr packet.  If the
+   * address is found, set |*result| and return true.  Otherwise,
+   * return false.
+   */
+  bool get_tls_address(size_t offset, remote_ptr<void> load_module,
+                       remote_ptr<void>* result);
+
+  /**
+   * Indicate that the symbol |name| has the given address.
+   */
+  void register_symbol(const std::string& name, remote_ptr<void> address);
+
+  /**
+   * Return a set of the names of all the symbols that have not yet
+   * been resolved.
```

Comment:
Shouldn't we return the names of all symbols, whether or not they have been resolved? In case symbols already resolved need to be re-resolved to a different address?